scipy/linalg/tests/test_decomp.py

```diff
@@ -765,29 +769,79 @@ def test_eigh_tridiagonal(self):
+            temp = np.random.rand(self.n, self.n).astype(d)
+            temp = temp + temp.T
+            self.testmats_a += [temp.copy()]
+            if label in 'cz':
+                temp = np.random.rand(self.n, self.n).astype(d)*d(1j)
+                temp = temp + temp.conj().T
+                self.testmats_a[-1] += temp
+
+            temp = np.random.rand(self.n, self.n).astype(d)
+            temp = temp + temp.T
+            self.testmats_b += [temp + np.eye(self.n, dtype=d)*d(10.)]
+            if label in 'cz':
+                temp = np.random.rand(self.n, self.n).astype(d)*d(1j)
+                temp = temp + temp.conj().T
+                # make sure b is pos. def.
+                self.testmats_b[-1] += temp + np.eye(self.n, dtype=d)*d(10.)
+
+    def test_wrong_inputs(self):
+        assert_raises(ValueError, eigh, np.ones([1, 2]))
+        assert_raises(ValueError, eigh, np.ones([2, 2]), np.ones([2, 1]))
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([2, 2]))
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([3, 3]),
+                      subset_by_value=[1, 2], eigvals=[2, 4])
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([3, 3]),
+                      eigvals=[0, 4])
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([3, 3]),
+                      eigvals=[2, 0])
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([3, 3]),
+                      subset_by_value=[2, 0])
+        assert_raises(ValueError, eigh, np.ones([2, 2]), driver='wrong')
+        assert_raises(ValueError, eigh, np.ones([3, 3]), None, driver='gvx')
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([3, 3]),
+                      driver='evr', turbo=False)
+        assert_raises(ValueError, eigh, np.ones([3, 3]), np.ones([3, 3]),
+                      driver='gvd', eigvals=[1, 2], turbo=False)
+
+    def test_nonpositive_b(self):
+        assert_raises(LinAlgError, eigh, np.ones([3, 3]), np.ones([3, 3]))
+
+    # index based subsets are done in the legacy test_eigh()
+    def test_value_subsets(self):
+        for ind, dt in enumerate(DTYPES):
+            w, v = eigh(self.testmats_a[ind], subset_by_value=[-2, 2])
+            assert_equal(v.shape[1], len(w))
+            assert all((w > -2) & (w < 2))
+
+            w, v = eigh(self.testmats_a[ind],
+                        self.testmats_b[ind],
+                        subset_by_value=[-2, 2])
+            assert_equal(v.shape[1], len(w))
+            assert all((w > -2) & (w < 2))
+
+
+# Old eigh tests kept for backwards compatibility
+@pytest.mark.parametrize('eigvals', (None, (2, 4)))
+@pytest.mark.parametrize('turbo', (True, False))
+@pytest.mark.parametrize('lower', (True, False))
+@pytest.mark.parametrize('overwrite', (True, False))
+@pytest.mark.parametrize('dtype_', ('f', 'd', 'F', 'D'))
+@pytest.mark.parametrize('dim', (6,))
+def test_eigh(dim, dtype_, overwrite, lower, turbo, eigvals):
```

Comment:
Do all the drivers get tested?

Reply:
index based ones and regular ones are tested here and the value based ones are tested above. Didn't want to replicate things. But we can add more.

Reply:
Ah, I didn't see the new "driver" option used, and I don't remember the logic for how one is selected if unspecified.